openweathermap-ios/openweathermap-ios/Weather/Model/WeatherManager.swift

```diff
@@ -14,31 +14,31 @@ struct WeatherManager {
         getRequest(with: urlString)
     }
     
-    func getRequest(with utlString: String) {
-        if let url = URL(string: utlString) {
-            let session = URLSession(configuration: .default)
-            let task = session.dataTask(with: url) { (data, response, erro) in
-                if erro != nil{
-                    return
-                }
-                if let safeData = data {
-                    if let weather = self.parseJSON(safeData){
-                        self.delegate?.didUpdateWeather(self, weather: weather)
-                    }
+    private func getRequest(with utlString: String) {
+        guard let url = URL(string: utlString) else { return }
+        let session = URLSession(configuration: .default)
+        let task = session.dataTask(with: url) { (data, response, erro) in
+            if erro != nil{
+                return
+            }
+            if let safeData = data {
+                if let weather = self.parseJSON(safeData){
```

Comment:
Capture self to avoid retain cycles.

Reply:
I don't know what that means :(.

Reply:
https://medium.com/mackmobile/avoiding-retain-cycles-in-swift-7b08d50fe3ef